### PR TITLE
[BD-18] Fixes to run e2e tests on devstack

### PR DESCRIPTION
**Pull request by ziafazal**
Fixes to run e2e tests on devstack.

### CODE REVIEW

**Comment by ziafazal** [comment]
@feanil @kdmccormick could you please review this PR?

File: Makefile [hunk 1]
```diff
@@ -101,6 +101,10 @@ ifndef COMPOSE_FILE
 $(error FS_SYNC_STRATEGY is set to $(FS_SYNC_STRATEGY). Must be one of: local-mounts, nfs, docker-sync)
 endif
 
+ifndef COMPOSE_PROJECT_NAME
+COMPOSE_PROJECT_NAME = devstack
+endif
+
```
Comment (kdmccormick) [inline Makefile]: Does this have any effect? Ideally this should already be handled in [options.mk](https://github.com/edx/devstack/blob/master/options.mk#L35-L39).

File: Makefile [hunk 2]
```diff
@@ -578,10 +582,11 @@ _expects-database.%:
 ########################################################################################
 
 e2e-tests: ## Run the end-to-end tests against the service containers.
-	docker run -t --network=${COMPOSE_PROJECT_NAME:-devstack}_default -v ${DEVSTACK_WORKSPACE}/edx-e2e-tests:/edx-e2e-tests -v ${DEVSTACK_WORKSPACE}/edx-platform:/edx-e2e-tests/lib/edx-platform --env-file ${DEVSTACK_WORKSPACE}/edx-e2e-tests/devstack_env edxops/e2e env TERM=$(TERM)  bash -c 'paver e2e_test --exclude="whitelabel\|enterprise"'
+	docker run -t --network=$(COMPOSE_PROJECT_NAME)_default -v $(DEVSTACK_WORKSPACE)/edx-e2e-tests:/edx-e2e-tests --env-file $(DEVSTACK_WORKSPACE)/edx-e2e-tests/devstack_env edxops/e2e env TERM=$(TERM) bash -c 'paver e2e_test'
```
Comment (kdmccormick) [inline Makefile:581]: ~Regarding the `:-devstack` addition, I have the same comment as above about `options.mk`.~
Nevermind this comment; I read the diff backwards.

**Review by kdmccormick — COMMENTED** [review]
Thanks for the PR. I'm excited to see e2e tests working in Devstack again!

I only have a couple of comments; I haven't tested this. @feanil will you be reviewing this, or did you want me to do a fuller review?

**Comment by feanil** [comment]
@kdmccormick if you can do a further review, that would be great but I'm happy to take it if you're strapped for time.

File: Makefile (hunk 1 above)
Comment (kdmccormick) [reply Makefile]: I was able to get the tests working without this change. I believe that your other fixes are sufficient to get the tests running again.

File: Makefile [hunk 3]
```diff
@@ -578,10 +582,11 @@ _expects-database.%:
 ########################################################################################
 
 e2e-tests: ## Run the end-to-end tests against the service containers.
```
Comment (kdmccormick) [inline Makefile]:
```suggestion
e2e-tests: dev.up.lms+studio ## Run the end-to-end tests against the service containers.
```
How about making both this and `e2e-tests.with-shell` targets depend on `dev.up.lms+studio`? They break if those services haven't been brought up, right?

**Review by kdmccormick — CHANGES_REQUESTED** [review]
Generally looks good!

I was able to run tests, but the ones I tried to run have failed. For example:
```
root@fa406f207ae7:/edx-e2e-tests# paver e2e_test common/test_html_component.py
```
gives:
```
bok_choy.page_object.PageLoadError: Message: Timed out waiting to load page 'CourseOutlinePageExtended(<browser>)' at URL 'http://studio.devstack.edx:18010/course/course-v1:edX+E2E-101+course'
```
That being said, at least the tests run now, which is better than before, so I'm happy to give my :+1: on this once my comments are addressed.

**Comment by ziafazal** [comment]
@kdmccormick Thanks for reviewing. I have addressed feedback could you please take another look?

**Review by kdmccormick — APPROVED** [review]
Looks good, with the caveat that this just gets test to run for me, but not to pass.

**Comment by feanil** [comment]
Ecom failure looks like it's happening on master as well.

**Pull request merged**